pkg/providers/amifamily/resolver.go

```diff
@@ -134,25 +136,43 @@ func (r DefaultResolver) Resolve(nodeClass *v1.EC2NodeClass, nodeClaim *karpv1.N
 		// In order to support reserved ENIs for CNI custom networking setups,
 		// we need to pass down the max-pods calculation to the kubelet.
 		// This requires that we resolve a unique launch template per max-pods value.
-		// Similarly, instance types configured with EfAs require unique launch templates depending on the number of
+		// Similarly, instance types configured with EFAs require unique launch templates depending on the number of
 		// EFAs they support.
+		// Reservations IDs are also included since we need to create a separate LaunchTemplate per reservation ID when
+		// launching reserved capacity. If it's a reserved capacity launch, we've already filtered the instance types
+		// further up the call stack.
 		type launchTemplateParams struct {
-			efaCount int
-			maxPods  int
+			efaCount       int
+			maxPods        int
+			reservationIDs string
```

Comment:
Why is this a string and not a slice since we have to convert it to a slice for using it elsewhere anyways?

Reply:
To use this with `GroupBy` the object needs to be comparable, which a slice isn't. Since ordering doesn't matter in this case, combining this into a string works pretty naturally.